### ISSUE TITLE
failed to build customized caddy with forwardproxy

### ISSUE THREAD

**Issue by xh116** [klzgrad/naiveproxy#164]
```
~/go/bin/xcaddy build --with github.com/caddyserver/forwardproxy@caddy2=github.com/klzgrad/forwardproxy@naive
2021/02/19 11:22:10 [INFO] Resolved relative replacement github.com/caddyserver/forwardproxy@caddy2=github.com/klzgrad/forwardproxy@naive to /root/github.com/klzgrad/forwardproxy@naive
2021/02/19 11:22:10 [INFO] Temporary folder: /tmp/buildenv_2021-02-19-1122.922859870
2021/02/19 11:22:10 [INFO] Writing main module: /tmp/buildenv_2021-02-19-1122.922859870/main.go
2021/02/19 11:22:10 [INFO] Initializing Go module
2021/02/19 11:22:10 [INFO] exec (timeout=10s): /usr/local/go/bin/go mod init caddy 
go: creating new go.mod: module caddy
2021/02/19 11:22:10 [INFO] Replace github.com/caddyserver/forwardproxy => /root/github.com/klzgrad/forwardproxy@naive
2021/02/19 11:22:10 [INFO] exec (timeout=10s): /usr/local/go/bin/go mod edit -replace github.com/caddyserver/forwardproxy=/root/github.com/klzgrad/forwardproxy@naive 
2021/02/19 11:22:10 [INFO] Pinning versions
2021/02/19 11:22:10 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddyserver/caddy/v2 
go: errors parsing go.mod:
/tmp/buildenv_2021-02-19-1122.922859870/go.mod:5: malformed module path "/root/github.com/klzgrad/forwardproxy": empty path element
2021/02/19 11:22:10 [FATAL] exit status 1
```
 

Also tried with another module like `--with github.com/caddy-dns/cloudflare `  separately and build successfully.

problem happened under fresh installed debian 10 OS and fresh go environment. 

Since I am not a developer, so not sure what the problem is. 
Hope someone knows the resolution could do me a favor.

**Comment by klzgrad** [comment]
Can you build this with non-root and report again?

**Comment by xh116** [comment]
> Can you build this with non-root and report again?

Tested just now, but got the same error. 

-------------------------------------------------

But I found another solution by `git clone` the forked forwardproxy firstly, then compile. 

```

- git clone -b naive https://github.com/klzgrad/forwardproxy

- go get -u github.com/caddyserver/xcaddy/cmd/xcaddy
 
- ~/go/bin/xcaddy build --with github.com/caddyserver/forwardproxy=$PWD/forwardproxy

```

Anybody else encountered the same problem, the command above may help...

**Comment by klzgrad** [comment]
This should be a bug in xcaddy: https://github.com/caddyserver/xcaddy/issues/50.

**Comment by francislavoie** [comment]
FYI, will be fixed by https://github.com/caddyserver/xcaddy/pull/51 👍

**Comment by xh116** [comment]
> FYI, will be fixed by [caddyserver/xcaddy#51](https://github.com/caddyserver/xcaddy/pull/51) 👍

Latest xcaddy works well now. Thanks! 

```
~/go/bin/xcaddy build --with github.com/caddyserver/forwardproxy@caddy2=github.com/klzgrad/forwardproxy@naive
2021/02/27 01:44:01 [INFO] Temporary folder: /tmp/buildenv_2021-02-27-0144.780187251
2021/02/27 01:44:01 [INFO] Writing main module: /tmp/buildenv_2021-02-27-0144.780187251/main.go
2021/02/27 01:44:01 [INFO] Initializing Go module
2021/02/27 01:44:01 [INFO] exec (timeout=10s): /usr/local/go/bin/go mod init caddy 
go: creating new go.mod: module caddy
2021/02/27 01:44:01 [INFO] Replace github.com/caddyserver/forwardproxy => github.com/klzgrad/forwardproxy@naive
2021/02/27 01:44:01 [INFO] exec (timeout=10s): /usr/local/go/bin/go mod edit -replace github.com/caddyserver/forwardproxy=github.com/klzgrad/forwardproxy@naive 
2021/02/27 01:44:01 [INFO] Pinning versions
2021/02/27 01:44:01 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddyserver/caddy/v2 
go: github.com/caddyserver/caddy/v2 upgrade => v2.3.0
2021/02/27 01:44:14 [INFO] Build environment ready
2021/02/27 01:44:14 [INFO] Building Caddy
2021/02/27 01:44:14 [INFO] exec (timeout=0s): /usr/local/go/bin/go mod tidy 
go: downloading github.com/klzgrad/forwardproxy v0.0.0-20210120121422-9b4a5a242bd6
go: found github.com/caddyserver/forwardproxy in github.com/caddyserver/forwardproxy v0.0.0-00010101000000-000000000000
2021/02/27 01:44:18 [INFO] exec (timeout=0s): /usr/local/go/bin/go build -o /root/caddy -ldflags -w -s -trimpath 
2021/02/27 01:47:24 [INFO] Build complete: ./caddy
2021/02/27 01:47:24 [INFO] Cleaning up temporary folder: /tmp/buildenv_2021-02-27-0144.780187251
```

**Comment by klzgrad** [comment]
Thanks to @francislavoie for following up!

**Comment by Mr-Sheep** [comment]
Same problem when using `~/go/bin/xcaddy build --with github.com/caddyserver/forwardproxy@caddy2=github.com/klzgrad/forwardproxy@naive`. Tried on multiple servers

```
2021/04/11 09:37:48 [INFO] Resolved relative replacement github.com/caddyserver/forwardproxy@caddy2=github.com/klzgrad/forwardproxy@naive to /root/github.com/klzgrad/forwardproxy@naive
2021/04/11 09:37:48 [INFO] Temporary folder: /tmp/buildenv_2021-04-11-0937.114824196
2021/04/11 09:37:48 [INFO] Writing main module: /tmp/buildenv_2021-04-11-0937.114824196/main.go
2021/04/11 09:37:48 [INFO] Initializing Go module
2021/04/11 09:37:48 [INFO] exec (timeout=10s): /usr/local/go/bin/go mod init caddy
go: creating new go.mod: module caddy
go: to add module requirements and sums:
        go mod tidy
2021/04/11 09:37:48 [INFO] Replace github.com/caddyserver/forwardproxy => /root/github.com/klzgrad/forwardproxy@naive
2021/04/11 09:37:48 [INFO] exec (timeout=10s): /usr/local/go/bin/go mod edit -replace github.com/caddyserver/forwardproxy=/root/github.com/klzgrad/forwardproxy@naive
2021/04/11 09:37:48 [INFO] Pinning versions
2021/04/11 09:37:48 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddyserver/caddy/v2
go: errors parsing go.mod:
/tmp/buildenv_2021-04-11-0937.114824196/go.mod:5: malformed module path "/root/github.com/klzgrad/forwardproxy": empty path element
2021/04/11 09:37:48 [FATAL] exit status 1
```
One has to fork [klzgrad/forwardproxy](https://github.com/klzgrad/forwardproxy) and use `~/go/bin/xcaddy build --with github.com/caddyserver/forwardproxy=$PWD/forwardproxy` to build with xcaddy

Consider compile caddy with naive tweaked forwardproxy with naiveproxy releases?

**Comment by francislavoie** [comment]
@Mr-Sheep you need a build of `xcaddy` from master, not the latest v0.1.8 release. The bug was fixed in https://github.com/caddyserver/xcaddy/commit/83bc5a7c589a5c655a777ffd3197c24ee5506e90 which hasn't landed in a tagged release yet.

**Comment by francislavoie** [comment]
`v0.1.9` is tagged now, so you can use that: https://github.com/caddyserver/xcaddy/releases/tag/v0.1.9

**Comment by Mr-Sheep** [comment]
> @Mr-Sheep you need a build of `xcaddy` from master, not the latest v0.1.8 release. The bug was fixed in [caddyserver/xcaddy@83bc5a7](https://github.com/caddyserver/xcaddy/commit/83bc5a7c589a5c655a777ffd3197c24ee5506e90) which hasn't landed in a tagged release yet.

Thanks, I did not notice that.
Successfully built with xcaddy 0.1.9
